packages/core/src/components/RadioButton/RadioButtonGroup.tsx

```diff
@@ -17,31 +18,39 @@ const { Provider } = radioButtonGroupContext;
 const RadioButtonGroup: React.FC<RadioButtonGroupProps> = ({
   direction = Direction.Vertical,
   value,
-  onValueChange = () => {},
+  onValueChange,
   defaultValue,
   style,
   children,
   ...rest
 }) => {
-  const [internalValue, setInternalValue] = React.useState<string | undefined>(
-    value || defaultValue
-  );
+  const [internalValue, setInternalValue] = React.useState<
+    string | undefined
+  >();
 
   React.useEffect(() => {
-    if (value != null) {
-      setInternalValue(value);
+    const realValue = getRealValue(value);
+
+    if (realValue) {
+      setInternalValue(realValue);
     }
   }, [value]);
 
   React.useEffect(() => {
-    if (defaultValue != null) {
-      setInternalValue(defaultValue);
+    const realDefaultValue = getRealValue(defaultValue);
+
+    if (realDefaultValue) {
+      setInternalValue(realDefaultValue);
     }
   }, [defaultValue]);
 
-  const handleValueChange = (newValue: string) => {
-    setInternalValue(newValue);
-    onValueChange(newValue);
+  const handleValueChange = (newValue: any) => {
+    const realNewValue = getRealValue(newValue);
+
+    if (realNewValue) {
+      setInternalValue(newValue);
+      onValueChange?.(newValue);
```

Comment:
any reason you removed that default function to replace it with this?

Reply:
@peter It's just a safety to be sure the value is not undefined.

Reply:
@peter Good to go.